### ISSUE TITLE
hostname_replace has stopped working in latest 2024.8.8+dcf95644c

### ISSUE THREAD

**Issue by 825i** [searxng/searxng#3722]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**

**How did you install SearXNG?**
Through UNRAID
**What happened?**
I updated without problems but hostname_replace plugin isn't working in the latest version.

**How To Reproduce**
```
enabled_plugins:
#   # these plugins are enabled if nothing is configured ..
#   - 'Hash plugin'
#   - 'Self Information'
#   - 'Tracker URL remover'
#   - 'Ahmia blacklist'  # activation depends on outgoing.using_tor_proxy
#   # these plugins are disabled if nothing is configured ..
    - 'Hostname replace'  # see hostname_replace configuration below
#   - 'Open Access DOI rewrite'
#   - 'Tor check plugin'
#   # Read the docs before activate: auto-detection of the language could be
#   # detrimental to users expectations / users can activate the plugin in the
#   # preferences if they want.
#   - 'Autodetect search language'

# Configuration of the "Hostname replace" plugin:
#

hostname_replace:
#  '(.*\.)?facebook\.com$': false
#  '(.*\.)?instagram\.com$': false
#  '(.*\.)?quora\.com$': false
#  '(.*\.)?twitter\.com$': false
#  '(.*\.)?x\.com$': false
#   '(.*\.)?youtube\.com$': 'invidious.example.com'
#   '(.*\.)?youtu\.be$': 'invidious.example.com'
#   '(.*\.)?youtube-noocookie\.com$': 'yotter.example.com'
#   '(.*\.)?reddit\.com$': 'teddit.example.com'
#   '(.*\.)?redd\.it$': 'teddit.example.com'
#   '(www\.)?twitter\.com$': 'nitter.example.com'
#   # to remove matching host names from result list, set value to false
  '(.*\.)?facebook\.com$': false
  '(.*\.)?instagram\.com$': false
  '(.*\.)?quora\.com$': false
  '(.*\.)?twitter\.com$': false
  '(.*\.)?x\.com$': false
  '(.*\.)?reddit\.com$': false
  '(.*\.)?youtube\.com$': false
  '(.*\.)?google\.com$': false
  '(.*\.)?apple\.com$': false
  '(.*\.)?tripadvisor\.com$': false
  '(.*\.)?tripadvisor\.com.au$': false
  '(.*\.)?tripadvisor\.in$': false
  '(.*\.)?deviantart\.com$': false
  '(.*\.)?shutterstock\.com$': false
  '(.*\.)?researchgate\.net$': false
```

**Expected behavior**
The above sites that are not commented out should not show as results in my instance. However, they do still form many of the search engine results even though they should have been removed.

**Additional context**
The above is part of my `settings.yml`, yet it seems SearxNG is not respecting the hostname_replace plugin any longer.

**Comment by return42** [comment]
- see https://github.com/searxng/searxng/discussions/3553#discussioncomment-10273806

**Comment by 825i** [comment]
> * see [hostname_replace has been droped by PR #3552 #3553 (comment)](https://github.com/searxng/searxng/discussions/3553#discussioncomment-10273806)

I'm getting this error when trying to enable the new "hostnames" plugin:

```
2024-08-10 11:20:45,494 ERROR:searx.plugins.hostnames: Exception while calling on_result
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 203, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/hostnames.py", line 136, in on_result
    result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/__init__.py", line 334, in _compile_template
    return _sre.template(pattern, _parser.parse_template(repl, pattern))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 999, in parse_template
    s = Tokenizer(source)
        ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 235, in __init__
    string = str(string, 'latin1')
             ^^^^^^^^^^^^^^^^^^^^^
TypeError: decoding to str: need a bytes-like object, bool found
2024-08-10 11:20:45,494 ERROR:searx.plugins.hostnames: Exception while calling on_result
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 203, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/hostnames.py", line 136, in on_result
    result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/__init__.py", line 334, in _compile_template
    return _sre.template(pattern, _parser.parse_template(repl, pattern))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 999, in parse_template
    s = Tokenizer(source)
        ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 235, in __init__
    string = str(string, 'latin1')
             ^^^^^^^^^^^^^^^^^^^^^
TypeError: decoding to str: need a bytes-like object, bool found
2024-08-10 11:20:45,495 ERROR:searx.plugins.hostnames: Exception while calling on_result
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 203, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/hostnames.py", line 136, in on_result
    result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/__init__.py", line 334, in _compile_template
    return _sre.template(pattern, _parser.parse_template(repl, pattern))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 999, in parse_template
    s = Tokenizer(source)
        ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 235, in __init__
    string = str(string, 'latin1')
             ^^^^^^^^^^^^^^^^^^^^^
TypeError: decoding to str: need a bytes-like object, bool found
2024-08-10 11:20:45,495 ERROR:searx.plugins.hostnames: Exception while calling on_result
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 203, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/hostnames.py", line 136, in on_result
    result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/__init__.py", line 334, in _compile_template
    return _sre.template(pattern, _parser.parse_template(repl, pattern))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 999, in parse_template
    s = Tokenizer(source)
        ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 235, in __init__
    string = str(string, 'latin1')
             ^^^^^^^^^^^^^^^^^^^^^
TypeError: decoding to str: need a bytes-like object, bool found
2024-08-10 11:20:45,496 ERROR:searx.plugins.hostnames: Exception while calling on_result
Traceback (most recent call last):
  File "/usr/local/searxng/searx/plugins/__init__.py", line 203, in call
    ret = getattr(plugin, plugin_type)(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/searxng/searx/plugins/hostnames.py", line 136, in on_result
    result[parsed] = result[parsed]._replace(netloc=pattern.sub(replacement, result[parsed].netloc))
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/__init__.py", line 334, in _compile_template
    return _sre.template(pattern, _parser.parse_template(repl, pattern))
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 999, in parse_template
    s = Tokenizer(source)
        ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/re/_parser.py", line 235, in __init__
    string = str(string, 'latin1')
             ^^^^^^^^^^^^^^^^^^^^^
TypeError: decoding to str: need a bytes-like object, bool found
```

**Comment by 825i** [comment]
I solved the above by changing in `settings.yml`

```
hostnames:
  replace: 'rewrite-hosts.yml'
```

to

```
hostnames:
  remove: 'rewrite-hosts.yml'
```

**Comment by return42** [comment]
Yeah, read the docs  https://docs.searxng.org/dev/plugins/hostnames.html and this https://github.com/searxng/searxng/issues/3558#issuecomment-2175058128 gives a good example

**Comment by 825i** [comment]
It would be nice to have a bit of information in the official docs that indicates hostnames_replace is depreciated:

https://docs.searxng.org/src/searx.plugins.hostnames.html

**Comment by return42** [comment]
> It would be nice to have a bit of information in the official docs that indicates hostnames_replace is depreciated

Since June we have a announcement about backward compatibility https://github.com/searxng/searxng/discussions/categories/announcements ..

![image](https://github.com/user-attachments/assets/cedc422f-43b3-4336-b0aa-3c1c1b490cff)

and we posted it several times at [#searxng:matrix.org](https://matrix.to/#/#searxng:matrix.org) and in the maintainer's Matrix room .. there are several issues here on GH talking about hostname_replace .. and TBH: does anyone really read the docs on issues / did you read the docs before posting this issue? :-)

BTW in the transition period there was a hint in the docs: https://github.com/searxng/searxng/pull/3552/files#diff-4198c1a8aa9a1d3cbff921013996b3c5ec70dcd3ce1550539f16e469d16e2620

**Comment by dennis1248** [comment]
The first thing I did was check the docs, the docs do not detail much. It was unclear what the new plugin was called and what the exact syntax is.

https://docs.searxng.org/dev/plugins/hostnames.html <- Does not clearly tell us the name of the plugin, and lacks any mention of this recent breaking change.

Only once checking the Gitub issues of other people complaining about it did I learn what exactly changed and how to configure it.

In the future consider adding a notice to relevant doc pages after something has been deprecated or saw a breaking change and provide some simple concise reference material on how to port the old config to the new one.


For other people who will also come across this issue;

```
# Old
enabled_plugins:
    - 'Hostname replace'

# New
enabled_plugins:
    - 'Hostnames plugin'
```

```
# Old
hostname_replace:
    '(.*\.)?youtube\.com$': 'invidious.example.com'

# New
hostnames:
    replace:
        '(.*\.)?youtube\.com$': 'invidious.example.com'
```

**Comment by return42** [comment]
> In the future consider adding a notice to relevant doc pages after something has been deprecated or saw a breaking change and provide some simple concise reference material on how to port the old config to the new one.

Nothing is so perfect that it can't be improved.  SearXNG is a community project, any contribution for improvement is welcome ... send us your PR / Thank you!

The documentation source can be found here:

https://github.com/searxng/searxng/blob/f1c05e7c16e59a8d5d98640bc057958ea268cddf/searx/plugins/hostnames.py#L3-L80
